### ISSUE TITLE
Sub - process /usr/libexec/cydia/cydo returned an error code (2)

### ISSUE THREAD

**Issue by surrel14** [checkra1n/BugTracker#119]
Seems an error with permissions of Cydo. It doesn’t let install/upgrade anything including Filza and OpenSSH, which would be useful to SSH into device and change permissions to the aforesaid Cydo file.

**Comment by nullpixel** [comment]
Were you jailbroken with unc0ver before?

**Comment by mattmalec** [comment]
I have the same error too. Yes, I was previously jailbroken with unc0ver.

**Comment by Yongle-Fu** [comment]
same too with unc0ver

**Comment by kpwn** [comment]
You can use the dropbear running on port 44 to add +s to cydo - we ship it precisely to solve these sorts of issues.

**Comment by mattmalec** [comment]
I'm not entirely certain how I would connect with that. It is similar to SSH?

**Comment by keithtp1** [comment]
Same issue here tried everything on google didn't find any fix

**Comment by kpwn** [comment]
It's ssh, it's just on a different port, and it only works via usb, not wifi.

**Comment by mattmalec** [comment]
Ahh, so it’d use Putty for example and connect with a COM port. Got it. Now which command would I run to add +s to cydo? Seems I’m unable to chmod it.

**Comment by keithtp1** [comment]
I ran this command “chmod 06555 /usr/libexec/cydia/cydo” and it did nothing. Even tried installing Filza to do it but no success

**Comment by kpwn** [comment]
you need to do chown 0:0 /usr/libexec/cydia/cydo ; chmod +s /usr/libexec/cydia/cydo

**Comment by mattmalec** [comment]
> I ran this command “chmod 06555 /usr/libexec/cydia/cydo” and it did nothing. Even tried installing Filza to do it but no success

Yeah, it doesn’t like when I try to chmod it. Says it’s either a read-only file or I can’t chmod the file.

**Comment by mattmalec** [comment]
> you need to do chown 0:0 /usr/libexec/cydia/cydo ; chmod +s /usr/libexec/cydia/cydo

Got it! Had to append `-f` to the end to force it to work, but it’s still giving me the same original error.

**Comment by kpwn** [comment]
the easiest solution would be to use the checkra1n loader app to restore the rootfs and install cydia cleanly.

**Comment by keithtp1** [comment]
Doesn't work tried that 10 times.

**Comment by keithtp1** [comment]
Any fixes yet?

**Comment by mattmalec** [comment]
> Doesn't work tried that 10 times.

You didn’t restore the RootFS then.

**Comment by keithtp1** [comment]
How do you do that? Its in the checkra1n app right?

**Comment by mattmalec** [comment]
You need to go on u0 and check the “Restore rootFS” option and then hit the “Jailbreak” button. Fairly certain that will get rid of your jailbreak and put it back in a state where you’re able to rejailbreak with checkra1n

**Comment by keithtp1** [comment]
I’m on ios 13.2.2 it doesn’t work.

**Comment by keithtp1** [comment]
fixed it! 

Step 1: First fully unjailbreak your phone
Step 2: Make a backup with itunes
Step 3: Factory reset your system and restore the backup
Step 4: Jailbreak with checkra1n
Step 5: Install Cydia
and boom it works!
